libraries/Microsoft.Bot.Builder.Azure/AzureTableStorage.cs

```diff
@@ -3,164 +3,369 @@
 
 using System;
 using System.Collections.Generic;
+using System.Dynamic;
 using System.Linq;
 using System.Net;
+using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
 using Microsoft.Bot.Builder.Core.Extensions;
 using Microsoft.WindowsAzure.Storage;
 using Microsoft.WindowsAzure.Storage.Table;
-using Newtonsoft.Json;
 
 namespace Microsoft.Bot.Builder.Azure
 {
     /// <summary>
-    /// Models IStorage around a dictionary 
+    /// Middleware that implements an Azure Table based storage provider for a bot.
     /// </summary>
     public class AzureTableStorage : IStorage
     {
-        private static HashSet<string> _checkedTables = new HashSet<string>();
-
+        /// <summary>
+        /// Underlying Azure Table.
+        /// </summary>
         public CloudTable Table { get; private set; }
```

Comment:
I wouldn't expose the table personally... it's an internal implementation detail. Consider using just use a `private readonly` field.